command/agent/command.go

```diff
@@ -1082,7 +1082,13 @@ func (c *Command) Run(args []string) int {
 	}
 
 	// Get the new client http listener addr
-	httpAddr, err := config.ClientListener(config.Addresses.HTTP, config.Ports.HTTP)
+	var httpAddr net.Addr
+	var err error
+	if config.Ports.HTTP != -1 {
+		httpAddr, err = config.ClientListener(config.Addresses.HTTP, config.Ports.HTTP)
+	} else {
+		httpAddr, err = config.ClientListener(config.Addresses.HTTPS, config.Ports.HTTPS)
```

Comment:
I think it would, we should have it error out if they've got both http and http disabled.